solidity/contracts/BondedECDSAKeepFactory.sol

```diff
@@ -480,6 +480,24 @@ contract BondedECDSAKeepFactory is
         return tokenStaking.balanceOf(_operator);
     }
 
+    /// @notice Gets the total weight of operators
+    /// in the sortition pool for the given application.
+    /// @dev Reverts if sortition does not exits for the application.
+    /// @param _application Address of the application.
+    /// @return The sum of all registered operators' weights in the pool.
+    /// If no pool exists, returns 0.
+    function getSortitionPoolWeight(
+        address _application
+    ) public view returns (uint256) {
+        address poolAddress = candidatesPools[_application];
+
+        if (poolAddress == address(0)) {
+            return 0;
+        }
```

Comment:
I think we should rather revert if the pool does not exist:
```
require(
            candidatesPools[_application] != address(0),
            "No pool found for the application"
        );
```

We'll have this behavior consistent with `getSortitionPool`.

Reply:
Let's also make sure to convert the two possible paths in this function in unit tests: https://github.com/keep-network/keep-ecdsa/blob/master/solidity/test/BondedECDSAKeepFactoryTest.js

Reply:
Reverting if the pool doesn't exist can cause some problems with the load balancer strategy if the pool hasn't been created. If we're only using it in an application-specific way this is fine. Also, the `getSortitionPoolWeight` function doesn't necessarily need to be exposed here and we can query `getSortitionPool`, handling errors there if relevant, and then directly call the sortition pool to get total weight.

Reply:
> Reverting if the pool doesn't exist can cause some problems with the load balancer strategy if the pool hasn't been created. If we're only using it in an application-specific way this is fine.

I think this is what we expect. My intuition is that the load balancer is an application-specific strategy, it has a reference to two ECDSA keep factories and the load balancer should expect those factories to be initialized. If they are not, it means something is wrong.

Regarding `getSortitionPoolWeight`, I kinda like we return just `uin256` because the application does not have to import `SortitionPool` that is kinda internal to how keep is doing group selection. Do you think just the weight is not going to be enough?

Reply:
The weight is enough in realistic situations. There are some cases (whales with lots of KEEP but not much bondable ETH) where the weight would give false positives wrt the ability to select KEEP-backed members but those would be hard to account for anyway and we can just treat them as situations where you won't be able to create a new keep and fail in some reasonable way.

Reply:
Yep and the good part of having the load balancing strategy on the application side is that the application may `CALL` the selected factory and if that call fails, it may fallback to the other factory if that's what the application wants to do. We are not doing any assumptions about that on Keep side.